dashboard/src/src/__tests__/VisualizationsPage.test.js

```diff
@@ -1,135 +1,115 @@
+    await page.goto('localhost:3000');
   });
-});
 
-test('Test if we switch to visualization page', async () => {
-  // Load page
-  render(<App />, { wrapper: MemoryRouter });
-  // Let time for ExperimentNavBar to load experiments
-  // to prevent warnings about async calls not terminated
-  expect(
-    await screen.findByText(/2-dim-shape-exp/, {}, global.CONFIG_WAIT_FOR_LONG)
-  ).toBeInTheDocument();
+  test('Test if we switch to visualization page', async ({ page }) => {
+    // Let time for ExperimentNavBar to load experiments
+    const firstExperiment = await page.getByText(/2-dim-shape-exp/);
+    await firstExperiment.waitFor();
+    await expect(firstExperiment).toHaveCount(1);
 
-  // Make sure we are on default (landing) page
-  expect(screen.queryByText(/Landing Page/)).toBeInTheDocument();
+    // Make sure we are on default (landing) page
+    await expect(await page.getByText(/Landing Page/)).toHaveCount(1);
 
-  // Make sure we are not on visualizations page
-  expect(screen.queryByText(/Nothing to display/)).toBeNull();
+    // Make sure we are not on visualizations page
+    await expect(await page.getByText(/Nothing to display/)).toHaveCount(0);
 
-  // Get visualizations page link
-  const menu = screen.queryByTitle(/Go to experiments visualizations/);
-  expect(menu).toBeInTheDocument();
+    // Go to visualization page
 
-  // CLick on visualizations page link
-  fireEvent.click(menu);
+    const menuExperiments = await page.locator('nav > ul > li:nth-child(1)');
+    await expect(menuExperiments).toHaveCount(1);
+    await expect(menuExperiments).toBeVisible();
+    await menuExperiments.click();
+    const menu = await menuExperiments.getByTitle(
+      /Go to experiments visualizations/
+    );
+    await expect(menu).toHaveCount(1);
+    await expect(menu).toBeVisible();
+    await menu.click();
 
-  // Check we are on visualizations page
-  const elements = await screen.findAllByText(/Nothing to display/);
-  expect(elements.length).toBe(3);
-});
+    // Check we are on visualizations page
+    const elements = await page.getByText(/Nothing to display/);
+    await expect(elements).toHaveCount(3);
+  });
 
-test('Test if we can select and unselect experiments', async () => {
-  // Load page
-  render(<App />, { wrapper: MemoryRouter });
-  const experiment = await screen.findByText(
-    /2-dim-shape-exp/,
-    {},
-    global.CONFIG_WAIT_FOR_LONG
-  );
-  expect(experiment).toBeInTheDocument();
+  test('Test if we can select and unselect experiments', async ({ page }) => {
+    const firstExperiment = await page.getByText(/2-dim-shape-exp/);
+    await firstExperiment.waitFor();
 
-  // Switch to visualizations page
-  const menu = screen.queryByTitle(/Go to experiments visualizations/);
-  fireEvent.click(menu);
-  expect((await screen.findAllByText(/Nothing to display/)).length).toBe(3);
+    // Go to visualization page
+    const menuExperiments = await page.locator('nav > ul > li:nth-child(1)');
```

Comment:
Are the elements more difficult to find using playwright?

Reply:
No, it is pretty simple ! It was more difficult with puppeteer, and that's why I searched for another framework.

The difference at that line is that, previously, we were in a simulated document structure, I which we could locate an element and "click" on it even if element is hidden, as long as it's in the document structure, like a sub-menu.

But now that we are in a real rendered page, it seems I cannot click on an hidden element. Following code just waits indefinitely until test timeout:

```javascript
    // Go to visualization page
    const menu = await page.getByTitle(
      /Go to experiments visualizations/
    );
    await menu.click();  // waits indefinitely here
```
So, I need to first click on top-menu, to make sub-menu appear, and then click to visible sub-menu.